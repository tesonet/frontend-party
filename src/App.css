.login {
  background: url(./img/background.png) no-repeat center center fixed;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-color: rgba(0, 0, 0, 0);
}

body {
  background-color: #0b0f27;
  font-family: 'Roboto', sans-serif;
}


.App {
  text-align: center;
}

.App-logo {
  width: 246px;
  pointer-events: none;
}

.App-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}