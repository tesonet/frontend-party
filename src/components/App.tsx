import { hot } from 'react-hot-loader/root';
import React from 'react';
import {
  BrowserRouter,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { Normalize } from 'styled-normalize';
import { ThemeProvider } from 'styled-components';
import GlobalStyle from './styled/GlobalStyle';
import Login from './pages/Login';
import ProtectedRoute from './utils/ProtectedRoute';
import ServerList from './pages/ServerList';
import theme from '../theme';
import { Provider } from 'react-redux';
import store from '../store';
import CheckIfLoggedIn from './utils/CheckIfLoggedIn';

function App() {
  return (
    <Provider store={store}>
      <CheckIfLoggedIn>
        <ThemeProvider theme={theme}>
          <Normalize />
          <GlobalStyle />
          <BrowserRouter>
            <Switch>
              <ProtectedRoute exact path="/" component={ServerList} />
              <Route exact path="/login" component={Login} />
              <Redirect to="/" />
            </Switch>
          </BrowserRouter>
        </ThemeProvider>
      </CheckIfLoggedIn>
    </Provider>
  );
}

export default hot(App);
