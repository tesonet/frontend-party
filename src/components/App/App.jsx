import React from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'
import Login from '../../containers/Login/Login';
import List from '../../containers/List/List';
import NotFound from '../NotFound/NotFound';
import PrivateRoute from '../PrivateRoute/PrivateRoute';

const App = () => {
  return (
    <Router>
      <Switch>
        {["/", "/login"].map(path => (
          <Route
            key={path}
            exact
            path={path}
            component={Login}
          />
        ))}
        <PrivateRoute exact path="/list" component={List} />
        <Route path="*" component={NotFound} />
      </Switch>
    </Router>
  )
}

export default App;