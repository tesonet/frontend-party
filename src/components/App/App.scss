.App {
  background-image: url('../assets/images/Background.png');
  background-position: center right;
  background-size: cover;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}