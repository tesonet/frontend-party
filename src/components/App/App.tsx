import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
  RouteProps
} from "react-router-dom";
import Login from "../Login/Login";
import Servers from "../Servers/Servers";
import { Provider, useSelector } from "react-redux";
import store, { State } from "../../store/store";

const PrivateRoute: React.FC<RouteProps> = ({
  component: Component,
  ...rest
}) => {
  const token = useSelector(({ user }: State) => (user ? user.token : null));

  return (
    <Route
      {...rest}
      render={props =>
        token ? <Component {...props} /> : <Redirect to="/login" />
      }
    />
  );
};

const App: React.FC = () => (
  <Router>
    <Provider store={store}>
      <Switch>
        <Route exact path="/" render={() => <Redirect to="/login" />} />
        <Route path="/login" component={Login} />
        <PrivateRoute path="/servers" component={Servers} />
      </Switch>
    </Provider>
  </Router>
);

export default App;
