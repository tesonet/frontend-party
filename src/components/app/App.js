import React, { Component } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Dashboard from '../dashboard/Dashboard';
import SignIn from '../auth/SignIn';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Switch>
            <Route exact path='/' component={Dashboard} />
            <Route path='/signin' component={SignIn} />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
  
}

export default App;