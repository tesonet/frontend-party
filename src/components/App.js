import React from 'react';
import { Router, Route } from 'react-router-dom';
import history from './history';
import LoginPage from './Login/LoginPage';
import ResultPage from './Result/ResultPage';
import '../sass/app.scss';

const App = () => {
  return(
    <div className="new-app">
      <Router history={history}>
        <div>
          <Route path="/" exact component={LoginPage} />
          <Route path="/result" exact component={ResultPage} />
        </div>
      </Router>
    </div>
  );
};

export default App;
