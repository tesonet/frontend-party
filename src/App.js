import React from 'react'
import {Provider} from 'react-redux'
import {Router, Route, Switch} from 'react-router-dom'

import {configureStore} from './store'
import {Login, Dashboard, NotFound} from './components/pages'
import PrivateRoute from './components/common/PrivateRoute'
import CssBaseline from './ui/CssBaseline'
import './App.css'

const {store, history} = configureStore()

function App() {
  return (
    <Provider store={store}>
      <CssBaseline />
      <Router history={history}>
        <Switch>
          <PrivateRoute exact path="/" component={Dashboard} />
          <Route exact path="/login" component={Login} />
          <Route path="*" component={NotFound} />
        </Switch>
      </Router>
    </Provider>
  )
}

export default App
