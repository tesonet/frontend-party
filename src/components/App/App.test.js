import * as React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

configure({ adapter: new Adapter() });
it('should render component correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});