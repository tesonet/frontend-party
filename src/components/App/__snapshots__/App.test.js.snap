// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      key="/"
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      key="/login"
      path="/login"
    />
    <PrivateRoute
      component={[Function]}
      exact={true}
      path="/list"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</BrowserRouter>
`;
